packages/docusaurus-theme-classic/src/theme/BlogPostItem/Header/Author/index.tsx

```diff
@@ -22,8 +27,36 @@ export default function BlogPostItemHeaderAuthor({
   author,
   className,
 }: Props): JSX.Element {
-  const {name, title, url, imageURL, email} = author;
+  const {name, title, url, socials, imageURL, email} = author;
+  const {github, twitter, stackoverflow, linkedin} = socials || {};
   const link = url || (email && `mailto:${email}`) || undefined;
+  const renderSocialMedia = () => (
+    <div className={clsx(styles.authorSocial, 'avatar__subtitle')}>
+      {twitter && (
+        <Link href={twitter}>
+          <Twitter className={clsx(styles.socialIcon)} />
+        </Link>
+      )}
```

Comment:
This is a bit duplicated.

What you could do instead is put all icons in a map.

```tsx
const PlatformIconsMap: Record<string,ComponentType<{className: string}>> = {
  twitter: Twitter,
  // ...
}

function PlatformLink({platform,link}) {
  const Icon = PlatformIconsMap[platform] ?? DefaultIcon;
  return (
        <Link href={link}>
          <Icon className={clsx(styles.socialIcon)} />
        </Link>
  )
}

Object.entries(socials ?? {}).map(([platform,link]) => {
  return <PlatformLink key={platform} platform={platform} link={link}/>;
})
```

We also probably want to preserve the order in which social platforms were provided by the user. If they are primarily YouTubers, they probably want the youtube icon to appear first, instead of always having Twitter first like you did